pkg/api/defaults.go

```diff
@@ -649,11 +649,12 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool) {
 			if profile.VMSSOverProvisioningEnabled == nil {
 				profile.VMSSOverProvisioningEnabled = to.BoolPtr(DefaultVMSSOverProvisioningEnabled && !isUpgrade && !isScale)
 			}
-			if profile.Count > 100 {
-				profile.SinglePlacementGroup = to.BoolPtr(false)
-			}
+
 			if profile.SinglePlacementGroup == nil {
 				profile.SinglePlacementGroup = to.BoolPtr(DefaultSinglePlacementGroup)
+				if profile.Count > 100 {
+					profile.SinglePlacementGroup = to.BoolPtr(false)
```

Comment:
do we want to set to false if the cluster is BLB?

Reply:
the master branch behavior seems more accurate:
https://github.com/Azure/aks-engine/blob/master/pkg/api/defaults.go#L546

Reply:
I think it is right to consistent with master branch. removing count> 100 logic, instead of adding SLB checking logic.